evalml/pipelines/utils.py

```diff
@@ -1418,7 +1418,7 @@ def unstack_multiseries(
         for column_name in full_dataset.columns.drop([time_index, series_id]):
             new_column = single_series[column_name]
             new_column.index = new_time_index
-            new_column.name = f"{column_name}_{s_id}"
+            new_column.name = f"{column_name}|{s_id}"
```

Comment:
Yea, I think seeing this change here and later on makes me wonder if we should have a an importable constant like `SEPARATOR_SYMBOL` so that they can all be changed in the same place?